unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarModifiers/TriggerArea.cs

```diff
@@ -12,11 +13,10 @@ public class BoxTriggerArea : TriggerArea
 {
     public Vector3 box;
 
-    public override Collider AddCollider(GameObject avatarModifierArea)
+    public override GameObject[] DetectAvatars(Vector3 center, Quaternion rotation)
     {
-        BoxCollider boxCollider = avatarModifierArea.AddComponent<BoxCollider>();
-        boxCollider.isTrigger = true;
-        boxCollider.size = box;
-        return boxCollider;
+        Collider[] colliders = Physics.OverlapBox(center, box / 2, rotation, LayerMask.GetMask("AvatarTriggerDetection"), QueryTriggerInteraction.Collide);
+        return colliders.Select(collider => collider.transform.parent.gameObject)
+            .ToArray();
```

Comment:
I think this `TriggerArea` class is no longer needed and we could do this detection in the `AvatarModifierArea` instead, what do you think?

Besides that, I try to avoid using Linq as it comes with a performance cost, in this case the query will be called on every frame (and every avatar modifier area) so it would be better if we can avoid that usage here. If this class disappears and we move this OverlapBox to the `AvatarModifierArea`, we could just populate the hashset with the corresponding colliders instead of dealing with an array with Linq.